src/org/loklak/data/DAO.java

```diff
@@ -933,10 +933,19 @@ public static int deleteOld(IndexName indexName, Date createDateLimit) {
          * @param resultCount - the number of messages in the result; can be zero if only aggregations are wanted
          * @param aggregationLimit - the maximum count of facet entities, not search results
          * @param aggregationFields - names of the aggregation fields. If no aggregation is wanted, pass no (zero) field(s)
+         * @param filterList - list of filters in String datatype
          */
-        public SearchLocalMessages(final String q, final Timeline.Order order_field, final int timezoneOffset, final int resultCount, final int aggregationLimit, final String... aggregationFields) {
+        public SearchLocalMessages (
+                final String q,
+                final Timeline.Order order_field,
+                final int timezoneOffset,
+                final int resultCount,
+                final int aggregationLimit,
+                final ArrayList<String> filterList,
```

Comment:
Yes, `List` can't be converted to `ArrayList`. It's the other way round, as `ArrayList` is derived from `List`. I was talking about something like this - 
```java
List<String> someThing(List<String> list) {
    ArrayList<String> aList = (ArrayList) list;
    ...  // Do whatever needed
    return aList;
}
```